src/inc/sift-events/sift-events.php

```diff
@@ -851,8 +898,23 @@ public static function send() {
 			}
 
 			foreach ( self::$to_send as $entry ) {
+				// We need the original WC user ID to handle the decision locally.
+				$wccom_user_id = $entry['properties']['$user_id'] ?? null;
```

Comment:
There are many things that might not be right here.
- naming `$wccom_user_id`: this pluing will not always be on WCCCOM
- but especifically on WCCOM, $user_id will either be the WPCOM ID, or after [this PR](https://github.com/Automattic/gold/issues/558) it can be a prefix with the WCCOM ID.

You need to assume this is a user_id, and add a filter to allow the sidecar plugin to change it to the right user.